Tests/LoggerTests/Loggers/FileLoggerTests.swift

```diff
@@ -9,146 +9,134 @@
 import XCTest
 
 class FileLoggerTests: XCTestCase {
+    private var userDefaults: UserDefaults!
+    private var fileManager: FileManager!
+    private var fileLoggerManager: FileLoggerManager!
 
     override func setUp() {
         super.setUp()
 
-        LogManager.shared.removeAllLoggers()
+        userDefaults = UserDefaults(suiteName: "testUserDefaults")!
+        fileManager = FileManager.default
+        fileLoggerManager = try! FileLoggerManager(fileManager: fileManager, userDefaults: userDefaults, numberOfLogFiles: 3)
     }
 
-    func test_inicialization_of_FileLogger() {
-        // Set default values for all Logger properties and store them to UserDefaults
-        let fileLoggerManager = FileLoggerManager()
-        fileLoggerManager.resetPropertiesToDefaultValues()
-
-        if let currentLogFileNumber = UserDefaults.standard.object(forKey: Constants.UserDefaultsKeys.currentLogFileNumber) as? Int {
-            XCTAssertEqual(0, currentLogFileNumber)
-        } else {
-            XCTFail()
-        }
-
-        if let dateTimeOfLastLog = UserDefaults.standard.object(forKey: Constants.UserDefaultsKeys.dateOfLastLog) as? Date {
-            XCTAssertNotNil(dateTimeOfLastLog.toFullDateString().range(of: "^\\d{4}-\\d{2}-\\d{2}$", options: .regularExpression))
-        } else {
-            XCTFail()
-        }
-
-        if let numOfLogFiles = UserDefaults.standard.object(forKey: Constants.UserDefaultsKeys.numOfLogFiles) as? Int {
-            XCTAssertEqual(4, numOfLogFiles)
-        } else {
-            XCTFail()
-        }
-    }
+    override func tearDown() {
+        try! FileManager.default.removeItem(atPath: fileLoggerManager.logDirURL.path)
+        fileLoggerManager = nil
+        UserDefaults().removeSuite(named: "testUserDefaults")
```

Comment:
Použil bych spíš `UserDefaults.standard`. Nejsem si jistý, jak to přesně funguje, ale podle mě zavoláním `UserDefaults()` se vytvoří nějaký nový space, což v `tearDown()` není dle mého názoru ideální 😄.

Reply:
To uz je prepsany... UserDefaults.standard prave pouzivat nechci, to je neco jako Singleton pak...